src/ophyd_async/core/standard_readable.py

```diff
@@ -1,72 +1,168 @@
         return await merge_gathered_dicts(
-            [sig.describe() for sig in self._configuration_signals]
+            [sig.describe_configuration() for sig in self._configurables]
         )
 
     async def read_configuration(self) -> Dict[str, Reading]:
         return await merge_gathered_dicts(
-            [sig.read() for sig in self._configuration_signals]
+            [sig.read_configuration() for sig in self._configurables]
         )
 
     async def describe(self) -> Dict[str, Descriptor]:
-        return await merge_gathered_dicts(
-            [sig.describe() for sig in self._read_signals + self._read_uncached_signals]
-        )
+        return await merge_gathered_dicts([sig.describe() for sig in self._readables])
 
     async def read(self) -> Dict[str, Reading]:
-        return await merge_gathered_dicts(
-            [sig.read() for sig in self._read_signals]
-            + [sig.read(cached=False) for sig in self._read_uncached_signals]
-        )
+        return await merge_gathered_dicts([sig.read() for sig in self._readables])
+
+    @property
+    def hints(self) -> Hints:
+        return self._hints
+
+    @contextmanager
+    def add_children_as_readables(
+        self,
+        wrapper: Optional[Type[Union["ConfigSignal", "HintedSignal"]]] = None,
+    ) -> Generator[None, None, None]:
+        dict_copy = self.__dict__.copy()
+
+        yield
+
+        # Set symmetric difference operator gives all newly added items
+        new_attributes = dict_copy.items() ^ self.__dict__.items()
+        new_signals: List[SignalR] = [x[1] for x in new_attributes]
+
+        self._wrap_signals(wrapper, new_signals)
+
+    def add_readables(
+        self,
+        wrapper: Type[Union["ConfigSignal", "HintedSignal"]],
+        *signals: SignalR,
+    ) -> None:
+
+        self._wrap_signals(wrapper, signals)
+
+    def _wrap_signals(
+        self,
+        wrapper: Optional[Type[Union["ConfigSignal", "HintedSignal"]]],
+        signals: Sequence[SignalR],
+    ):
+
+        for signal in signals:
+            obj: Union[SignalR, "ConfigSignal", "HintedSignal"] = signal
+            if wrapper:
+                obj = wrapper(signal)
+
+            if isinstance(obj, AsyncReadable):
+                self._readables.append(obj)
+
+            if isinstance(obj, AsyncConfigurable):
+                self._configurables.append(obj)
+
+            if isinstance(obj, AsyncStageable):
+                self._stageables.append(obj)
+
+            if isinstance(obj, HasHints):
+                new_hint = obj.hints
+
+                # Merge the existing and new hints, based on the type of the value.
+                # This avoids default dict merge behaviour that overrided the values;
+                # we want to combine them when they are Sequences, and ensure they are
+                # identical when string values.
+                for key, value in new_hint.items():
+                    if isinstance(value, Sequence):
```

Comment:
I can't remember if str is a Sequence, if so then it needs to go first in the if statement

Reply:
You are correct, `str` is a `Sequence`. Changed the order.